### PR TITLE
minimal-printf: Enable by a target config parameter

### PR DESCRIPTION
**Pull request by hugueskamba**
### Pull request type

    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### CODE REVIEW

File: tools/toolchains/arm.py [hunk 1]
```diff
@@ -81,6 +81,12 @@ def __init__(self, target, notify=None, macros=None,
             if "--library_type=microlib" not in self.flags['common']:
                 self.flags['common'].append("--library_type=microlib")
 
+        if (
+            target.printf_lib == "minimal-printf" and
```
Comment (evedon) [inline tools/toolchains/arm.py]: I'd prefer these two lines joined together but that's a minor point.

**Review by evedon — COMMENTED** [review]
Great!

File: tools/toolchains/arm.py (hunk 1 above)
Comment (hugueskamba) [reply tools/toolchains/arm.py]: It makes the line too long not very Pythonic. It's sorta standard to keep lines under 79 chars https://black.readthedocs.io/en/stable/the_black_code_style.html#line-length

Comment (evedon) [reply tools/toolchains/arm.py]: I meant Lines 84 & 85

Comment (hugueskamba) [reply tools/toolchains/arm.py]: The popular formatter "Black" would format it more or less as I did.
```bash
$ black -l 79 test_python.py --diff
reformatted test_python.py
All done! ✨ 🍰 ✨
--- test_python.py      2019-11-14 14:51:46.563480 +0000
+++ test_python.py      2019-11-14 14:51:52.001901 +0000
@@ -1,2 +1,6 @@
-if target.printf_lib == "minimal-printf" and "-DMBED_MINIMAL_PRINTF" not in self.flags['common']:
+if (
+    target.printf_lib == "minimal-printf"
+    and "-DMBED_MINIMAL_PRINTF" not in self.flags["common"]
+):
     pass
```

**Comment by hugueskamba** [comment]
A PR has been raised upstream https://github.com/ARMmbed/mbed-os/pull/11891

**Pull request closed without merging**